cmd/opentelemetry/app/exporter/kafkaexporter/kafka_exporter.go

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2020 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafkaexporter
+
+import (
+	"context"
+
+	"github.com/spf13/viper"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/config/configtls"
+	"go.opentelemetry.io/collector/exporter/kafkaexporter"
+
+	"github.com/jaegertracing/jaeger/cmd/opentelemetry/app/receiver/kafkareceiver"
+	"github.com/jaegertracing/jaeger/plugin/storage/kafka"
+)
+
+// TypeStr defines exporter type.
+const TypeStr = "kafka"
+
+// Factory wraps kafkaexporter.Factory and makes the default config configurable via viper.
+// For instance this enables using flags as default values in the config object.
+type Factory struct {
+	// Wrapped is kafka exporter
+	Wrapped component.ExporterFactory
+	// Viper is used to get configuration values for default configuration
+	Viper *viper.Viper
+}
+
+var _ component.ExporterFactory = (*Factory)(nil)
+
+// Type returns the type of the exporter.
+func (f Factory) Type() configmodels.Type {
+	return f.Wrapped.Type()
+}
+
+// CreateDefaultConfig returns default configuration of Factory.
+// This function implements OTEL component.ExporterFactoryBase interface.
+func (f Factory) CreateDefaultConfig() configmodels.Exporter {
+	cfg := f.Wrapped.CreateDefaultConfig().(*kafkaexporter.Config)
+
+	// InitFromViper fails if certain fields are not set.  Setting them here
+	//  to prevent the process from exiting.
+	f.Viper.Set("kafka.producer.required-acks", "local")
+	f.Viper.Set("kafka.producer.compression", "none")
+
+	opts := &kafka.Options{}
+	opts.InitFromViper(f.Viper)
+
+	cfg.Encoding = kafkareceiver.MustOtelEncodingForJaegerEncoding(opts.Encoding)
```

Comment:
`MustOtelEncodingForJaegerEncoding` returns `jaeger_proto` or `jaeger_json`. so the exporter can't use OTEL encodings.

Reply:
My bad, it can use all encoding supported by the wrapped instance. The OTEL config changes the configuration after the default one is created.